index.js

```diff
@@ -108,6 +127,31 @@ function outputChangelog (argv, cb) {
   })
 }
 
+function getNextVersion (release) {
+  var newVersion = semver.inc(pkg.version, release.releaseAs)
+
+  if (argv.preRelease) {
+    // hack: if a feat is commited after two patches it wants to increase
+    // the prerelease tag instead of the minor and start from 0
+    // the next 2 lines and the if statement should fix this
+    var realVersion = semver.inc(pkg.version, release.releaseAs)
+    var prereleaseVersion = semver.inc(pkg.version, 'prerelease', argv.tag)
+
+    if (semver.gt(realVersion, prereleaseVersion)) {
+      newVersion = semver.inc(pkg.version, release.releaseAs)
+      newVersion = newVersion + '-' + argv.tag + '.0'
+
+      if (semver.gt(prereleaseVersion, newVersion)) {
```

Comment:
@bcoe: Yes, this is really a little bit tricky here. It is also really hard to explain, but let me try.

This is a little bit hacky =(, because semver is not acting like expected. If you are at version 1.0.0 and commit(fix) and create a prerelease, semver gives you 1.0.1-beta.0, which is right.

If you add another commit now with a feature, and you create another prerelease it should become 1.1.0-beta.0. The problem is, semver gives you 1.0.1-beta.1 back.

So, line 140 is checking for this. If semver originally wants to create a bigger version (major or minor instead of patch) we want to use the  newVersion and append the tag + build it 0.

The second line is to ensure, that a new commit would result in an increase of the build id.

When I just see the code, id does not really make sense for me too. I removed the lines and tested it. Maybe this could be rewritten somehow for better understanding? Tests are checking this.